src/widget.js

```diff
@@ -77,14 +77,14 @@ var Widget = Base.$extend({
         // wEvents
         this._registerWEvents(["show", "hide"]);
 
+        // Name must be set before other properties (e.g. needed when setting children)
+        if (!this._name) {
```

Comment:
I'm not sure wether the `if` is needed here. It seems to me that it will always be true (that `this._name` does not exists yet)

Reply:
If a class inheriting from widget as something like this in the constructor it does 

```
__init__(params) {
    this.name = "test";
    this.$super(params);
}
```

Reply:
You're right !